cmd/neofs-node/config.go

```diff
@@ -790,13 +790,18 @@ func initLocalStorage(c *cfg) {
 		tombstone.WithTombstoneSource(tombstoneSrc),
 	)
 
+	var shardsAttached int
 	for _, optsWithMeta := range c.shardOpts() {
 		id, err := ls.AddShard(append(optsWithMeta.shOpts, shard.WithTombstoneSource(tombstoneSource))...)
-		fatalOnErr(err)
-
-		c.log.Info("shard attached to engine",
-			zap.Stringer("id", id),
-		)
+		if err != nil {
+			c.log.Error("failed to attach shard to engine", zap.Error(err))
+		} else {
+			shardsAttached++
+			c.log.Info("shard attached to engine", zap.Stringer("id", id))
+		}
+	}
+	if shardsAttached == 0 {
```

Comment:
is it discussed behavior? i mean could a node starting with just an error in the logs but without some planned shards be expected by an admin?

Reply:
Currently we are still dropping shards in `Init` , so this behaviour is not new.
In our model shard is an (almost) independent domain of failure. I am mostly thinking about automatic node restart after hardware failures. We can discuss this in future, I am not sure what the expected behaviour here is: for 11 shards repetitve config manipulations is a laborious and error-prone task.

Reply:
Or do you mean that we could start it in a degraded mode if we cannot write ID to the metabase?

Reply:
> so this behaviour is not new

but we had `fatalOnErr(err)` previously if a shard has not been attached. as i understand, that was a _very different_. losing shards without some huge "you are losing shards" banner bothers me but i dont have any good ideas of how to solve that

Reply:
> In our model shard is an (almost) independent domain of failure.

also, can't fully agree with this. one shard could contain `Lock`s, `Tombstone`s that protect/remove data from another so losing one shard is not just losing some objects. but that is our general problem